src/internal/connector/graph/collections.go

```diff
@@ -13,9 +13,14 @@ import (
 
 var _ data.BackupCollection = emptyCollection{}
 
+// TODO: move this out of graph.  /data would be a much better owner
+// for a generic struct like this.  However, support.StatusUpdater makes
+// it difficult to extract from this package in a generic way.
 type emptyCollection struct {
```

Comment:
thoughts on leaving the name `prefixCollection`?  `empty` makes it sound like it doesn't hold any data at all.  Not just a lack of folders.  This can make it difficult to interpret its usage intent.